vector/any.go

```diff
@@ -21,3 +23,24 @@ type Puller interface {
 }
 
 type Builder func(*zcode.Builder) bool
+
+func Combine(vec Any, index []uint32, add Any) Any {
+	var tags []uint32
+	var vecs []Any
+	size := int(vec.Len())
```

Comment:
I think you have to include index here.
```suggestion
	size := int(vec.Len()) + len(index)
```

Reply:
There's more to this here and I'm not exactly sure what to do about this.

Reply:
In that case, I think you should restore the desiredLen parameter to Combine since that version worked, and then we can maybe try removing it again later on.

Reply:
On second thought your suggestion was right but I needed to make adjustments to how combine worked.